plugins/modules/remote_management/stacki/stacki_host.py

```diff
@@ -163,33 +188,25 @@ def __init__(self, module):
                        'Content-type': 'application/json',
                        'Cookie': login_req.headers.get('Set-Cookie')}
 
-    def do_request(self, module, url, payload=None, headers=None, method=None):
-        res, info = fetch_url(module, url, data=payload, headers=headers, method=method)
+    def do_request(self, url, payload=None, headers=None, method=None):
+        res, info = fetch_url(self.module, url, data=payload, headers=headers, method=method)
 
         if info['status'] != 200:
             self.module.fail_json(changed=False, msg=info['msg'])
 
         return res
 
     def stack_check_host(self):
-        res = self.do_request(self.module, self.endpoint, payload=json.dumps({"cmd": "list host"}), headers=self.header, method="POST")
-
-        if self.hostname in res.read():
-            return True
-        else:
-            return False
+        res = self.do_request(self.endpoint, payload=json.dumps({"cmd": "list host"}), headers=self.header, method="POST")
+        return self.hostname in res.read()
```

Comment:
Sounds like a bug waiting to happen (`myhost` is assumed to be present when `myhostname` exists). But that's nothing we can easily solve, I guess, without knowing what the API actually returns...

Reply:
Yeah, this module needs a lot of TLC, but yes, a deeper knowledge of the API would be required - I am not planning on that.